include/natalie/random_value.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <utility>
+
+#include "natalie/class_value.hpp"
+#include "natalie/forward.hpp"
+#include "natalie/global_env.hpp"
+#include "natalie/integer_value.hpp"
+#include "natalie/value.hpp"
+#include <random>
+
+namespace Natalie {
+
+using namespace TM;
+
+class RandomValue : public Value {
+public:
+    RandomValue(ClassValue *klass)
+        : Value { Value::Type::Random, klass } { }
+
+    RandomValue()
+        : Value { Value::Type::Random, GlobalEnv::the()->Random() } { }
+
+    virtual void visit_children(Visitor &visitor) override final {
+        Value::visit_children(visitor);
+    }
+
+    ValuePtr initialize(Env *, ValuePtr);
+
+    ValuePtr rand(Env *, ValuePtr);
+    ValuePtr seed() { return new IntegerValue { m_seed }; }
+
+    virtual void gc_inspect(char *buf, size_t len) const override {
+        snprintf(buf, len, "<Random %p seed=%lld>", this, m_seed);
+    }
+
+private:
+    nat_int_t m_seed;
+    std::mt19937 m_generator;
```

Comment:
@richardboehme let's change this into a pointer, allocate it in the constructor, and deallocate it in a (to-be-added) destructor. That way we don't need to have the GC overhead of those huge Cell size Allocators.

```suggestion
    std::mt19937 *m_generator;
```

I tested dynamically allocating it in a simple cpp program and it seemed to work:

```cpp
#include <random>   
#include <stdio.h>                                                     
                                                                       
int main() {                                                           
    auto *mt_rand = new std::mt19937(time(0));                 
    printf("rand = %lu\n", (*mt_rand)());                                                                                                                                                                          
    return 0;                                                                                                                                                                                                      
}
```

Reply:
That's a great idea! 👍